logos/src/logos/queue/priority_queue.py

```
@@ -97,367 +83,248 @@ def enqueue(
-        """Move all queued entries for a model from one provider to another.
+            model_queues = self._queues.get(model_id)
+            if not model_queues:
+                return False
+            for queue in model_queues.values():
+                for _neg_pri, _ts, _eid, entry in queue:
+                    if entry.is_cold_at_queue:
+                        return True
+            return False
 
-        Returns the number of entries moved. Thread-safe.
+    def get_total_depth_by_provider(self, provider_id: int = None) -> int:
```

Ruff (0.15.12)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)